fx.go

```diff
@@ -19,3 +19,149 @@
 // THE SOFTWARE.
 
 package fx // import "go.uber.org/fx"
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
```

Comment:
Wait. We don't support having multiple `make(outbound)` right now without dispatcher being available, but we can totally do this -  

```go
func (yarpc) Constructors() []fx.Component {
    return []fx.Component {
    func(cfg config.Provider) *outbound1, *outbound2 {
        if cfg.outbounds {
                // make outbounds
                return outbound1, outbound2
            }
        }  
    }, 
    func(cfg config.Provider) *yarpc.Dispatcher {
        if cfg.inbounds {
            return dispatcher;
        }
    }
    }
}
```